### ISSUE TITLE
Let users default to VMware Fusion if desired

### ISSUE THREAD

**Issue by nathanleclaire** [docker/toolbox#107]
cc @jeffdm @mchiang0610 

Currently the Toolbox is tightly coupled to Virtualbox, however users with existing installs of VMware Fusion would like to be able to use that instead.  I am in favor of enabling this and Docker Machine has a driver so it's worth considering, although [there are some known issues with folder sharing](https://github.com/docker/machine/issues/1460) at the moment.

**Comment by mchiang0610** [comment]
@nathanleclaire I really like the idea, and want to support it 100%. That being said, right now the failure rate on local installs of VirtualBox is too high. If possible, I'd like us to focus on getting the experience smooth first for over 80% of our users. 

What do you think? 

cc @jeffdm @Elesant

**Comment by nathanleclaire** [comment]
Yeah, I agree to do Virtualbox right first, but wanted to document the issue for later reference, and make sure that the feature request didn't get lost in the noise.

**Comment by mchiang0610** [comment]
@nathanleclaire got it. Thanks!

**Comment by mchiang0610** [comment]
To support more than VMWare. That being said, this is not currently on the roadmap. Not our highest priority.

**Comment by gtirloni** [comment]
I'm wondering if there are any updates about supporting VMware Workstation on Windows.

Right now it seems a driver [exists](https://github.com/pecigonzalo/docker-machine-vmwareworkstation), but the Docker Toolbox installer insists on installing VirtualBox (not possible to disable that option).

**Comment by mlham** [comment]
I am still interested in an answer please.

**Comment by gtirloni** [comment]
With Docker for Windows now using Hyper-V, is it expected any work will be done on Docker Toolbox?

For users of VMware Workstation, neither the new Docker for Windows nor the Docker Toolbox seem like good options at this moment.

**Comment by gtirloni** [comment]
I was able to get Docker Toolbox working with VMware Workstation 12.1 on Windos 10.

# Install Docker Toolbox without VirtualBox:

DockerToolbox-<version>.exe /COMPONENTS="Docker,DockerMachine"

# Replace contents of  "C:\Program Files\Docker Toolbox\start.sh" with this:

	#!/bin/bash

	export PATH="$PATH:/mnt/c/Program Files (x86)/VMware/VMware Workstation"

	trap '[ "$?" -eq 0 ] || read -p "Looks like something went wrong in step ´$STEP´... Press any key to continue..."' EXIT

	VM=${DOCKER_MACHINE_NAME-default}
	DOCKER_MACHINE=./docker-machine.exe

	BLUE='\033[1;34m'
	GREEN='\033[0;32m'
	NC='\033[0m'


	if [ ! -f "${DOCKER_MACHINE}" ]; then
	  echo "Docker Machine is not installed. Please re-run the Toolbox Installer and try again."
	  exit 1
	fi

	vmrun.exe list | grep \""${VM}"\" &> /dev/null
	VM_EXISTS_CODE=$?

	set -e

	STEP="Checking if machine $VM exists"
	if [ $VM_EXISTS_CODE -eq 1 ]; then
	  "${DOCKER_MACHINE}" rm -f "${VM}" &> /dev/null || :
	  rm -rf ~/.docker/machine/machines/"${VM}"
	  #set proxy variables if they exists
	  if [ -n ${HTTP_PROXY+x} ]; then
		PROXY_ENV="$PROXY_ENV --engine-env HTTP_PROXY=$HTTP_PROXY"
	  fi
	  if [ -n ${HTTPS_PROXY+x} ]; then
		PROXY_ENV="$PROXY_ENV --engine-env HTTPS_PROXY=$HTTPS_PROXY"
	  fi
	  if [ -n ${NO_PROXY+x} ]; then
		PROXY_ENV="$PROXY_ENV --engine-env NO_PROXY=$NO_PROXY"
	  fi  
	  "${DOCKER_MACHINE}" create -d vmwareworkstation $PROXY_ENV "${VM}"
	fi

	STEP="Checking status on $VM"
	VM_STATUS="$(${DOCKER_MACHINE} status ${VM} 2>&1)"
	if [ "${VM_STATUS}" != "Running" ]; then
	  "${DOCKER_MACHINE}" start "${VM}"
	  yes | "${DOCKER_MACHINE}" regenerate-certs "${VM}"
	fi

	STEP="Setting env"
	eval "$(${DOCKER_MACHINE} env --shell=bash ${VM})"

	STEP="Finalize"
	clear
	cat << EOF


							##         .
					  ## ## ##        ==
				   ## ## ## ## ##    ===
			   /"""""""""""""""""\___/ ===
		  ~~~ {~~ ~~~~ ~~~ ~~~~ ~~~ ~ /  ===- ~~~
			   \______ o           __/
				 \    \         __/
				  \____\_______/

	EOF
	echo -e "${BLUE}docker${NC} is configured to use the ${GREEN}${VM}${NC} machine with IP ${GREEN}$(${DOCKER_MACHINE} ip ${VM})${NC}"
	echo "For help getting started, check out the docs at https://docs.docker.com"
	echo
	cd

	docker () {
	  MSYS_NO_PATHCONV=1 docker.exe "$@"
	}
	export -f docker

	if [ $# -eq 0 ]; then
	  echo "Start interactive shell"
	  exec "$BASH" --login -i
	else
	  echo "Start shell with command"
	  exec "$BASH" -c "$*"
	fi

# Docker Machine driver for VMware Workstation

Grab the executable from https://github.com/pecigonzalo/docker-machine-vmwareworkstation and place it in "C:\Program Files\Docker Toolbox"

Now run the Docker Quickstart Terminal and it should be okay.

**Comment by MichelDiz** [comment]
Can I do this on Mac OS Sierra? I 'm able to run the VirtualBox. However, Docker Tools does not recognize it - Some bug. And neither Docker for Mac is installing, even though I having a virtual machine running.

I have VMware Fusion. I have Bootcamp with windows 10. I have two Linux virtual machines. And I have Genymotion running some Android images in Virtualbox.

Is there any way to do this? to Generate a Docker Machine in VMware Fusion?

Cheers
